client-rest/src/app/transfer/initiate-transfer.component.ts

```diff
@@ -40,37 +42,49 @@ export class InitiateTransferComponent implements OnInit {
         let pollAttempt = 0;
         let maxPollAttempts = 20;
         let pollForWorkerKey = Observable.interval(1500)
-          .switchMap(() => this.transferService.getReservedWorker({id: this.progressService.transferId()}))
-          .take(maxPollAttempts)
-          .subscribe(
-            reservedWorker => {
-                console.log("polling for assigned transfer worker");
-                pollAttempt++;
-                if (reservedWorker.publicKey) {
-                  // TODO: Remove debug statement in production
-                  console.log("got transfer worker with public key: " + reservedWorker.publicKey);
-                  pollForWorkerKey.unsubscribe();
-                  this.progressService.workerReserved(reservedWorker.publicKey);
-                  this.startTransferJob();
-                  // TODO: encrypt creds with worker public key
-                } else if (pollAttempt == maxPollAttempts) {
-                  alert(`Timed out getting a worker for this data transfer`);
-                }
-            }, transportError);
+            .switchMap(() => this.transferService.getReservedWorker({id: this.progressService.transferId()}))
+            .take(maxPollAttempts)
+            .subscribe(
+                reservedWorker => {
+                    console.log("polling for assigned transfer worker");
+                    pollAttempt++;
+                    if (reservedWorker.publicKey) {
+                        pollForWorkerKey.unsubscribe();
+                        this.progressService.workerReserved(reservedWorker.publicKey);
+                        this.encryptAndStartTransfer();
+                    } else if (pollAttempt == maxPollAttempts) {
+                        alert(`Timed out getting a worker for this data transfer`);
+                    }
+                }, transportError);
+    }
+
+    encryptAndStartTransfer() {
+        // encrypt the export/import auth data pair as a JWE using the public key associated with the transfer job
+        let cryptographer = new Jose.WebCryptographer();
+        (<any>cryptographer).setContentEncryptionAlgorithm("A128CBC-HS256"); // workaround missing method definition in Typescript type definition for WebCryptographer
+        cryptographer.setKeyEncryptionAlgorithm("RSA-OAEP");
+        let serializedKey = JSON.parse(this.progressService.workerPublicKey());
+        Jose.Utils.importRsaPublicKey(serializedKey, "RSA-OAEP").then(rsaKey => {
+            let encrypter = new JoseJWE.Encrypter(cryptographer, rsaKey);
+            // encrypt the auth data
+            let authData = JSON.stringify({exportAuthData: this.progressService.exportAuthData(), importAuthData: this.progressService.importAuthData()});
```

Comment:
One of the nice things about JWE is that the RSA-OAEP scheme it does exactly what you outlined using a symmetric key and then transmits the encrypted symmetric key as part of the JWE in a defined format. This describes the flow:

https://tools.ietf.org/html/rfc7516#page-32

Since there are JWE libraries for a variety of languages, this will allow the server components to interop with clients written in other languages.

I added a comment in the code to point to the IETF doc.